File: backend/app/schemas/kebun.py
from pydantic import BaseModel
from typing import Any, Dict, List, Optional


class BlokProperties(BaseModel):
    id: int
    kebun: Optional[str] = None
    kode_blok: Optional[str] = None
    no_polygon: Optional[str] = None
    afdeling: Optional[str] = None
    komoditi: Optional[str] = None
    status: Optional[str] = None
    thn_tanam: Optional[str] = None
    varietas: Optional[str] = None
    kabupaten: Optional[str] = None
    kecamatan: Optional[str] = None
    desa: Optional[str] = None
    l_gis: Optional[float] = None
    l_rkap: Optional[float] = None
    l_hgu: Optional[float] = None
    protas_21: Optional[float] = None
    protas_22: Optional[float] = None
    protas_23: Optional[float] = None
    protas_24: Optional[float] = None
    populasi: Optional[int] = None
    pica: Optional[str] = None
    alas_hak: Optional[str] = None
    status_kpm: Optional[str] = None


class KebunStats(BaseModel):
    kebun: Optional[str] = None
    jumlah_blok: int
    total_luas: float


class KomoditiStats(BaseModel):
    komoditi: Optional[str] = None
    jumlah_blok: int
    total_luas: float


class StatsResponse(BaseModel):
    total_blok: int
    total_luas_gis: float
    total_luas_rkap: float
    per_kebun: List[Dict[str, Any]]
    per_komoditi: List[Dict[str, Any]]

File: frontend/types/kebun.ts
export interface BlokProperties {
  id: number;
  kebun: string | null;
  kode_blok: string | null;
  no_polygon: string | null;
  afdeling: string | null;
  komoditi: string | null;
  status: string | null;
  thn_tanam: string | null;
  varietas: string | null;
  kabupaten: string | null;
  kecamatan: string | null;
  desa: string | null;
  l_gis: number | null;
  l_rkap: number | null;
  l_hgu: number | null;
  protas_21: number | null;
  protas_22: number | null;
  protas_23: number | null;
  protas_24: number | null;
  populasi: number | null;
  pica: string | null;
  alas_hak: string | null;
  kat_als_hk: string | null;
  status_kpm: string | null;
  nomor_peta: string | null;
  keterangan: string | null;
  [key: string]: any;
}

export interface GeoJSONFeature {
  type: 'Feature';
  id: number;
  geometry: {
    type: string;
    coordinates: unknown;
  };
  properties: BlokProperties;
}

export interface FeatureCollection {
  type: 'FeatureCollection';
  features: GeoJSONFeature[];
  total: number;
}

export interface KebunStats {
  kebun: string;
  jumlah_blok: number;
  total_luas: number;
}

export interface StatsResponse {
  total_blok: number;
  total_luas_gis: number;
  total_luas_rkap: number;
  per_kebun: KebunStats[];
  per_komoditi: { komoditi: string; jumlah_blok: number; total_luas: number }[];
  per_status: { status: string; jumlah_blok: number }[];
}
